examples/Migration/CampaignManagement/CreateCompleteCampaignGoogleAdsApiOnly.php

```diff
@@ -0,0 +1,499 @@
+            $ad = new Ad([
+                'expanded_text_ad' => $expandedTextAdInfo,
+                'final_urls' => [new StringValue(['value' => 'http://www.example.com'])]
+            ]);
+
+            // Creates an ad group ad to hold the above ad.
+            $adGroupAd = new AdGroupAd([
+                'ad_group' => $adGroupResourceName,
+                'status' => AdGroupAdStatus::PAUSED,
+                'ad' => $ad
+            ]);
+
+            // Creates an ad group ad operation and add it to the operations array.
+            $adGroupAdOperation = new AdGroupAdOperation();
+            $adGroupAdOperation->setCreate($adGroupAd);
+            $operations[] = $adGroupAdOperation;
+        }
+
+        // Issues a mutate request to add the ad group ads.
+        $adGroupAdServiceClient = $googleAdsClient->getAdGroupAdServiceClient();
+        $adGroupAdResponse = $adGroupAdServiceClient->mutateAdGroupAds($customerId, $operations);
+
+        $newAdResourceNames = [];
+        foreach ($adGroupAdResponse->getResults() as $result) {
+            $newAdResourceNames[] = $result->getResourceName();
+        }
+
+        $newAds = self::getAds($googleAdsClient, $customerId, $newAdResourceNames);
+        foreach ($newAds as $newAd) {
+            printf(
+                "Created expanded text ad with id '%d', status %d and headline '%s - %s'.%s",
+                $newAd->getAd()->getId()->getValue(),
+                $newAd->getStatus(),
+                $newAd->getAd()->getExpandedTextAd()->getHeadlinePart1()->getValue(),
+                $newAd->getAd()->getExpandedTextAd()->getHeadlinePart2()->getValue(),
+                PHP_EOL
+            );
+        }
+    }
+
+    /**
+     * Gets an array of ads by their resource names.
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @param string $customerId the client customer ID without hyphens
+     * @param array $adResourceNames the resource names of the ads to retrieve
+     * @return array an array of ads
+     */
+    private static function getAds(
+        GoogleAdsClient $googleAdsClient,
+        string $customerId,
+        array $adResourceNames
+    ) {
+        $resourceNames = join(",", array_map(function ($resourceName) {
+            return "'$resourceName'";
+        }, $adResourceNames));
+        $googleAdsServiceClient = $googleAdsClient->getGoogleAdsServiceClient();
+        $query = "SELECT ad_group_ad.ad.id, " .
+                 "ad_group_ad.ad.expanded_text_ad.headline_part1, " .
+                 "ad_group_ad.ad.expanded_text_ad.headline_part2, " .
+                 "ad_group_ad.status, ad_group_ad.ad.final_urls, " .
+                 "ad_group_ad.resource_name from ad_group_ad where " .
+                 "ad_group_ad.resource_name in ($resourceNames)";
+
+        $response =
+            $googleAdsServiceClient->search($customerId, $query, ['pageSize' => self::PAGE_SIZE]);
+
+        $ads = [];
```

Comment:
Is this function (https://www.php.net/manual/en/function.iterator-to-array.php) can be used for this job?

Reply:
Not sure about it because it would also require using `array_map` to call `getAdGroupAd` on all items in the array. IMHO it's more readable this way, WDYT?

Reply:
OK. If it needs array_map, then this version is fine.